frontend/src/metabase/admin/settings/components/widgets/GroupMappingsWidget/GroupMappingsWidget.unit.spec.tsx

```diff
@@ -0,0 +1,171 @@
+import React from "react";
+import { act, render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import nock from "nock";
+
+import GroupMappingsWidget from "./GroupMappingsWidget";
+
+const setup = ({
+  mappingSetting = "ldap-group-mappings",
+  onChange = jest.fn(),
+  onChangeSetting = jest.fn(),
+  onSuccess = jest.fn(),
+  setting = { value: true },
+} = {}) => {
+  render(
+    <GroupMappingsWidget
+      mappingSetting={mappingSetting}
+      setting={setting}
+      onChangeSetting={onChangeSetting}
+      onSuccess={onSuccess}
+    />,
+  );
+};
+
+describe("GroupMappingsWidget", () => {
+  afterEach(() => {
+    nock.cleanAll();
+  });
+
+  describe("creating a mapping", () => {
+    beforeEach(() => {
+      nock(location.origin).get("/api/setting").reply(200, []);
+      nock(location.origin).get("/api/permissions/group").reply(200, []);
+
+      nock(location.origin).put("/api/setting/ldap-group-mappings").reply(200);
+    });
+
+    it("adds mapping to table on success", async () => {
+      setup();
+
+      userEvent.click(screen.getByText("New mapping"));
+      const input = screen.getByLabelText("new-group-mapping-name-input");
+      userEvent.type(input, "cn=Group");
+      act(() => {
+        screen.getByRole("button", { name: "Add" }).click();
```

Comment:
It would be nice to use `userEvent` here as well

Reply:
Oh yes, done 👍